provider/win_utils.py

```diff
@@ -0,0 +1,166 @@
+import os
+
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest import error_context, utils_misc
+from virttest.utils_windows import virtio_win
+
+
+@error_context.context_aware
+def get_viofs_exe(test, params, session):
+    """
+    Get viofs.exe from virtio win iso,such as E:\viofs\2k19\amd64
+    """
+    error_context.context("Get virtiofs exe full path.", test.log.info)
+    media_type = params["virtio_win_media_type"]
+    try:
+        get_drive_letter = getattr(virtio_win, "drive_letter_%s" % media_type)
+        get_product_dirname = getattr(virtio_win,
+                                      "product_dirname_%s" % media_type)
+        get_arch_dirname = getattr(virtio_win, "arch_dirname_%s" % media_type)
+    except AttributeError:
+        test.error("Not supported virtio win media type '%s'", media_type)
+    viowin_ltr = get_drive_letter(session)
+    if not viowin_ltr:
+        test.error("Could not find virtio-win drive in guest")
+    guest_name = get_product_dirname(session)
+    if not guest_name:
+        test.error("Could not get product dirname of the vm")
+    guest_arch = get_arch_dirname(session)
+    if not guest_arch:
+        test.error("Could not get architecture dirname of the vm")
+
+    exe_middle_path = ("{name}\\{arch}" if media_type == "iso"
+                       else "{arch}\\{name}").format(name=guest_name,
+                                                     arch=guest_arch)
+    exe_file_name = "virtiofs.exe"
+    exe_find_cmd = 'dir /b /s %s\\%s | findstr "\\%s\\\\"'
+    exe_find_cmd %= (viowin_ltr, exe_file_name, exe_middle_path)
+    exe_path = session.cmd(exe_find_cmd).strip()
+    test.log.info("Found exe file '%s'", exe_path)
+    return exe_path
+
+
+def viofs_svc_create(test, params, session, cmd):
+    """
+    Only for windows guest, to create a virtiofs service.
+
+    :param cmd: cmd to create virtiofs service
+    """
+    test.log.info("Register virtiofs service in Windows guest.")
+    exe_path = get_viofs_exe(test, params, session)
+    sc_create_s, sc_create_o = session.cmd_status_output(cmd % exe_path)
+    if sc_create_s != 0:
+        test.fail(
+            "Failed to register virtiofs service, output is %s" % sc_create_o)
+
+
+def viofs_svc_stop_start(test, params, session, action, cmd, expect_status):
```

Comment:
suggest to break this function into:
stop_viofs_svc
start_viofs_svc
query_viofs_svc
also add
delete_viofs_svc for we have a create_viofs_svc function